src/EditorFeatures/Test/QuickInfo/AbstractSemanticQuickInfoSourceTests.cs

```diff
@@ -257,36 +266,43 @@ internal Action<object> SymbolGlyph(Glyph expectedGlyph)
                 // var expectedTextCopy = "\r\n" + expectedText;
                 ClassificationTestHelper.Verify(expectedText, expectedClassifications, actualContent);
             };
+            */
         }
 
-        protected Action<object> NoTypeParameterMap
+        internal Action<QuickInfoItem> NoTypeParameterMap
         {
             get
             {
-                return (content) =>
+                return (qi) => AssertTaggedText(string.Empty, qi.TypeParameters);
+                /*
+                return (element) =>
                 {
+
                     Assert.Equal(string.Empty, ((QuickInfoDisplayDeferredContent)content).TypeParameterMap.ClassifiableContent.GetFullText());
                 };
+                */
             }
         }
 
-        protected Action<object> Usage(string expectedText, bool expectsWarningGlyph = false)
+        internal Action<QuickInfoItem> Usage(string expectedText, bool expectsWarningGlyph = false)
         {
-            return (content) =>
+            return (qi) => 
             {
-                var quickInfoContent = (QuickInfoDisplayDeferredContent)content;
-                Assert.Equal(expectedText, quickInfoContent.UsageText.ClassifiableContent.GetFullText());
-                Assert.Equal(expectsWarningGlyph, quickInfoContent.WarningGlyph != null && quickInfoContent.WarningGlyph.Glyph == Glyph.CompletionWarning);
+                AssertTaggedText(expectedText, qi.Usage);
+                if (expectsWarningGlyph)
+                {
+                    WarningGlyph(Glyph.CompletionWarning)(qi);
+                }
+                else
+                {
+                    Assert.False(qi.Tags.GetGlyphs().Contains(Glyph.CompletionWarning));
+                }
             };
         }
 
-        protected Action<object> Exceptions(string expectedText)
+        internal Action<QuickInfoItem> Exceptions(string expectedText)
         {
-            return (content) =>
-            {
-                var quickInfoContent = (QuickInfoDisplayDeferredContent)content;
-                Assert.Equal(expectedText, quickInfoContent.ExceptionText.ClassifiableContent.GetFullText());
-            };
+            return (qi) => AssertTaggedText(expectedText, qi.Exception);
```

Comment:
parens not necessary.  do not abbreviate to qi.  call it item, or quickInfo or something.